plugins/catalog-import/src/api/CatalogImportClient.ts

```diff
@@ -175,23 +176,33 @@ the component will become available.\n\nFor more information, read an \
     body: string;
   }): Promise<{ link: string; location: string }> {
     const { repositoryUrl, fileContent, title, body } = options;
-
-    const ghConfig = getGithubIntegrationConfig(
-      this.scmIntegrationsApi,
-      repositoryUrl,
-    );
-
-    if (ghConfig) {
-      return await this.submitGitHubPrToRepo({
-        ...ghConfig,
+    const parseData = YAML.parse(fileContent);
+    try {
+      const validationResponse = await this.catalogApi.validateEntity(
+        parseData,
+        `url:${repositoryUrl}`,
+      );
+      if (!validationResponse.valid) {
+        throw new Error(validationResponse.errors[0].message);
+      }
+      const ghConfig = getGithubIntegrationConfig(
+        this.scmIntegrationsApi,
         repositoryUrl,
-        fileContent,
-        title,
-        body,
-      });
-    }
+      );
 
-    throw new Error('unimplemented!');
+      if (ghConfig) {
+        return await this.submitGitHubPrToRepo({
+          ...ghConfig,
+          repositoryUrl,
+          fileContent,
+          title,
+          body,
+        });
+      }
+      throw new Error('unimplemented!');
+    } catch (err) {
+      throw err;
+    }
```

Comment:
If we're just catching to rethrow, do we need this block at all?

Reply:
yes both of them throw error no need for a try catch.